### DISCUSSION TITLE
Combining Tools and Knowledge

### DISCUSSION THREAD

**Discussion by quantsulting** [open-webui/open-webui#6248] [Q&A]
Hi everyone, I tried to provide some **knowledge** as a supplement to my standard model as well as a **tool** that allows you to analyze a stock using yahoo finance (tool available on the open-webui webpage). However, when I ask it, despite the tool's activation, to give me a report on a stock, it only searches in knowledge. how can i remedy this? thank you in advance for your suggestions. (N.B: it works If I remove the knowledge)

**Comment by Jet-lag** [comment]
With the tool and the supplement knowledge, is your model still using its own knowledge? I noticed that in my RAG application, if I use tool and supplement knowledge, the model no longer use its own knowledge. The base model I used is llama3.1:8b

**Comment by quantsulting** [comment]
Yes it does, I tested with Mistral and llama3.1.Le 17 oct. 2024 15:55, Jet-lag ***@***.***> a écrit :
With the tool and the supplement knowledge, is your model still using its own knowledge? I noticed that in my RAG application, if I use tool and supplement knowledge, the model no longer use its own knowledge. The base model I used is llama3.1:8b

**Comment by raghvender-1205** [comment]
Hi, is there any update on this ?.

I am using a tool which can retrieve user information, along with my model with a knowledge base. But, even after tool has been called. It still executes RAG to give me the result.

Is there any solution for this

**Reply by raghvender-1205** [reply]
@tjbck Can you help me with this ?

**Reply by quantsulting** [reply]
From my side I had to log out and log in again and it worked

**Reply by raghvender-1205** [reply]
In my setup, I have created model with a knowledge of some pdf files. I have also added a simple tool which would give me the user information. But, when I chat with this model and Ask it something like this

"Give me my details", it calls the tool displays the output but after that goes on to `generate_queries` and perform RAG on the prompt I have asked, and displays that output as well.

Is there any way to disable RAG after tool has been called ?
